### PR TITLE
Include the version of `go-github` in `User-Agent` headers sent to the GitHub API

### PR DESCRIPTION
**Pull request by timrogers**
👋🏻 Hi there! I'm Tim, and I'm a Product Manager at GitHub. To allow us to better understand how people are using our API, it's super helpful to know what versions of SDKs like `go-github` they are using. 

At the moment, this package sends the `User-Agent` "go-github" in requests, but it doesn't say what version is being used.

This updates the `User-Agent` header to include that information, whilst still maintaining the ability for people to choose their own user agent by setting the `UserAgent` field on `Client`.

It also exports a `Version` constant, so people can use that information - for example to construct custom `User-Agent` headers.

This will mean that someone has to update the `Version` constant with the new version when it is incremented.

This is the first ever Go I've written, so I won't be upset if you tell me that I've done this in a terrible way 😉

### CODE REVIEW

File: github/github.go
Comment (timrogers) [inline github/github.go:31]: I couldn't find this defined anywhere else in the package code, so I added it here. I'm happy to move it elsewhere, or access it in some other way if there are good options out there.

File: github/github.go
Comment (timrogers) [inline github/github.go:35]: What do you think of renaming this to `defaultUserAgent`, which is more representative of what it is?

File: github/github.go
Comment (timrogers) [inline github/github.go:172]: I thought that exposing this on the `Client` would help with (a) people constructing customer `User-Agent` headers and (b) me testing the default `User-Agent`!

File: github/github_test.go
Comment (timrogers) [inline github/github_test.go:517]: Is there somewhere else I should add a test to help confirm that this is being added to _all_ requests? Or do you think this is sufficient?

File: github/github.go
Comment (gmlewis) [inline github/github.go:308]: So this would no longer be needed since the `PackageVersion` constant is now exported and available.

File: github/github_test.go
Comment (gmlewis) [reply github/github_test.go:517]: I think this is sufficient, thanks.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Great idea, @timrogers !
Just a couple minor tweaks, please, then we should be ready for a second LGTM+Approval before merging.

File: github/github.go
Comment (timrogers) [inline github/github.go:31]: Any ideas of where I can document this to make sure it gets updated on release? I guess the "Maintainer's Guide" in `CONTRIBUTING.md` could be a good place.

**Review by timrogers — COMMENTED** [review]
Thanks for the feedback! I've addressed those comments and I think it's looking better now. Plus I learned about exported constants, which is great 🧠 ⬆️

File: github/github.go
Comment (gmlewis) [reply github/github.go:31]: How about we just change this to "Version" since that is pretty common?

Comment (gmlewis) [reply github/github.go:31]: Oh, and sure... feel free to add a comment to CONTRIBUTING.md, although I don't think we have any written-up instructions for maintainers yet.  Maybe I should do that sometime.  :smile:

Comment (gmlewis) [reply github/github.go:31]: Feel free to add a blank line after `Version` to uncouple it from the following unexported consts.

Comment (timrogers) [reply github/github.go]: Done! 🏁

**Review by gmlewis — APPROVED** [review]
Thank you, @timrogers !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this report before merging.

File: CONTRIBUTING.md
Comment (raynigon) [inline CONTRIBUTING.md:131]: This could be automated. Since the version constant is currently only used for the user agent, an error should not be a big problem, but if somebody depends on the constant this could lead to problems along the way.
Unfortunatly the release process in this project is not automated yet. If it gets automated in the future we need to include this constant in there.

**Review by raynigon — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @raynigon !
Merging.

**Pull request merged**
